torch/linalg/__init__.py

```diff
@@ -214,6 +214,86 @@
     (tensor(3.7417), tensor(11.2250))
 """)
 
+cond = _add_docstr(_linalg.linalg_cond, r"""
+linalg.norm(input, p=None, *, out=None) -> Tensor
+
+Computes the condition number of a matrix :attr:`input`,
+or of each matrix in a batched :attr:`input`, using the matrix norm defined by :attr:`p`.
+The condition number is defined as the matrix norm of
+:attr:`input` times the matrix norm of the inverse of :attr:`input`.
+
+This function supports ``float``, ``double``, and only on CPU, ``cfloat`` and ``cdouble`` dtype for :attr:`input`.
+
+.. note:: For `p = {None, 2, -2}` :attr:`input` can be non-square. For other norm types :attr:`input` must be
```

Comment:
this contradicts the above definition of "matrix norm of input times the matrix norm of the inverse of input", as non-square matrices don't have inverse.

Reply:
Should I add that for `p = {None, 2, -2}` pseudoinverse is used instead of inverse?

The condition number based on 2-norm is κ₂  = || A⁻¹ ||₂ ||A||₂. For j singular values, of A, that are ordered from largest to smallest with 1...j, ||A||₂ = σ₁ and || A⁻¹ ||₂ = 1 / σⱼ. Incidentally, we can compute singular values for non-square matrices, but the definition of the condition number stays the same.

Reply:
Yes, this is a helpful clarification (that I missed, thanks @ngimel!).

Reply:
I updated the note in 5ed8a77.
```
.. note:: For `p = {None, 2, -2}` there is a relation between singular value decomposition and matrix norm.
          For this case :func:`torch.linalg.svd` is used for computing the condition number as the ratio of
          the largest and smallest singular values. Since :func:`torch.linalg.svd` is used :attr:`input` can be non-square.
          For other norm types :attr:`input` must be a square matrix or a batch of square matrices.
          If :attr:`input` does not satisfy this requirement then a RuntimeError will be thrown.
```